### DISCUSSION TITLE
latest macOS runners need vcpkg support

### DISCUSSION THREAD

**Discussion by gblikas** [community/community#160649] [Actions]
[VCPKG](https://github.com/microsoft/vcpkg)  is a widely popular C++ dependency library, with minimal-setup caching support: https://learn.microsoft.com/en-us/vcpkg/consume/binary-caching-github-actions-cache. My opinion is that if [VCPKG](https://github.com/microsoft/vcpkg) has been removed from the macOS runner images, then Github should come out with `actions/setup-vcpkg`. 

The following macOS runners _**do not include** [VCPKG](https://github.com/microsoft/vcpkg) support_: 

- [macOS-15](https://github.com/actions/runner-images/blob/main/images/macos/macos-15-Readme.md)
- [macOS-15-arm64](https://github.com/actions/runner-images/blob/main/images/macos/macos-15-arm64-Readme.md)
- [macOS-14](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md)
- [macOS-14-arm64](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md)
- [macOS-13-arm64](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-arm64-Readme.md)

The following macOS runners _**do include** [VCPKG](https://github.com/microsoft/vcpkg) support_: 
- [macOS 13](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md)
- [macOS 12](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md)

**additional context**

- Linux and Windows have built-in VCPKG support. 
- `macos-latest` doesn't point to a valid macOS install; `macos-latest` should be a desktop install, like with `ubuntu-latest` and `windows-latest`

**Comment by vaishalipawar1490** [comment]
Hi @gblikas, 
Currently we are planning to keep the image size smaller, and prefer not to pre-install any item that could be installed during runtime. This would also reduce our maintenance concerns. 
Please continue to install vcpkg at runtime, as it is straightforward to do so.
If our decision changes we will keep you posted.

**Reply by gblikas** [reply]
>  planning to keep the image size smaller, and prefer not to pre-install any item that could be installed during runtime

@vaishalipawar1490 is vcpkg such a large dependency that it cannot be built into the images? It seems like its okay for Windows and Linux runner images (and some macOS images).
